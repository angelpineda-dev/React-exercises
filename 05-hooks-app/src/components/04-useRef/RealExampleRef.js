import { useState } from "react";
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks";

const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h2>Real Example Ref</h2>

      {show && <MultipleCustomHooks />}
      <button className="btn btn-secondary mt-2" onClick={() => setShow(!show)}>
        Show / Hide
      </button>
    </div>
  );
};

export default RealExampleRef;
