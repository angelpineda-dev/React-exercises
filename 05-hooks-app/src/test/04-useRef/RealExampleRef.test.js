import RealExampleRef from "../../components/04-useRef/RealExampleRef";
import { shallow } from "enzyme";

describe("Pruebas en <RealExampleRef/>", () => {
  const wrapper = shallow(<RealExampleRef />);

  test("Debe mostrarse correctamente. ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el componente <MultipleCustomHooks/>", () => {
    const btn = wrapper.find("button");

    btn.simulate("click");

    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
