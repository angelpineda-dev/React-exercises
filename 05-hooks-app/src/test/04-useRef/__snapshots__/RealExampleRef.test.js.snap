// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef/> Debe mostrarse correctamente.  1`] = `
<div>
  <h2>
    Real Example Ref
  </h2>
  <button
    className="btn btn-secondary mt-2"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
